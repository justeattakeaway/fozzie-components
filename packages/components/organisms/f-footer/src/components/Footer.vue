<template>
    <footer
        :data-theme="theme"
        class="c-footer"
        data-test-id="footer-component">
        <div
            class="c-footer-container c-footer-row c-footer-row--noPadBelowWide">
            <link-list
                v-for="(linkList, i) in copy.linkLists"
                :key="i + '_ButtonList'"
                :link-list="linkList" />
        </div>

        <div class="c-footer-light">
            <div class="c-footer-container">
                <div
                    v-if="copy.linkButtonList.length && showCourierLinks"
                    data-test-id="courierLinks-wrapper"
                    class="c-footer-row c-footer-row--noBottomPad">
                    <button-list
                        v-for="(buttonList, i) in copy.linkButtonList"
                        :key="i + '_ButtonList'"
                        :button-list="buttonList" />
                </div>

                <div class="c-footer-row">
                    <icon-list
                        :title="copy.downloadOurApps"
                        :icons="copy.appStoreIcons"
                        :locale="copy.locale"
                        is-apps />

                    <feedback-block
                        :title="copy.feedback"
                        :text="copy.improveOurWebsite"
                        :button-text="copy.sendFeedback" />

                    <icon-list
                        :icons="copy.socialIcons"
                        :title="copy.followUs"
                        is-social />
                </div>
            </div>
        </div>

        <div
            :class="['c-footer-container c-footer-row c-footer-row--combined c-footer-row--notEqualTopAndBottomPad c-footer-row--noPadBelowWide',
                     { 'c-footer-row--rightAlignedOnDesktopView': !showCountrySelector }]">
            <country-selector
                v-if="showCountrySelector"
                data-test-id="country-selector"
                :current-country-name="copy.currentCountryName"
                :current-country-key="copy.currentCountryKey"
                :countries="countryList"
                :change-country-text="copy.changeCurrentCountry" />

            <legal-field
                v-if="metaLegalFieldEnabled"
                :info="copy.metaLegalField" />

            <icon-list
                :icons="copy.paymentIcons"
                is-payments />
        </div>
    </footer>
</template>

<script>
import { globalisationServices } from '@justeat/f-services';
import ButtonList from './ButtonList.vue';
import CountrySelector from './CountrySelector.vue';
import FeedbackBlock from './FeedbackBlock.vue';
import IconList from './IconList.vue';
import LegalField from './LegalField.vue';
import LinkList from './LinkList.vue';
import { tenantConfigs, countries } from '../tenants';

export default {
    components: {
        ButtonList,
        CountrySelector,
        FeedbackBlock,
        IconList,
        LegalField,
        LinkList
    },
    props: {
        locale: {
            type: String,
            default: ''
        },

        showCourierLinks: {
            type: Boolean,
            default: true
        },
        showCountrySelector: {
            type: Boolean,
            default: true
        }
    },
    computed: {
        footerLocale () {
            return globalisationServices.getLocale(tenantConfigs, this.locale, this.$i18n);
        },
        copy () {
            const localeConfig = tenantConfigs[this.footerLocale];
            return localeConfig;
        },
        theme () {
            return globalisationServices.getTheme(this.footerLocale);
        },
        metaLegalFieldEnabled () {
            return Object.keys(this.copy.metaLegalField).length > 0;
        },
        countryList () {
            return countries.filter(country => country.key !== this.copy.currentCountryKey);
        }
    }
};
</script>

<style lang="scss">

$footer-heading-font-size: 'heading-s';

.c-footer {
    background-color: $footer-bgColor;
    color: $footer-textColor;
}

.c-footer-light {
    background-color: $footer-bgLight;
}

.c-footer-container {
    width: 100%;
    max-width: #{$layout-max-width}px;
    margin: 0 auto;
}

.c-footer-heading {
    @include font-size($footer-heading-font-size);
    font-family: $font-family-base;
    font-weight: $font-weight-headings;
    padding: spacing(x2);
    padding-left: 0;
}

.c-footer-heading--shortBelowWide {
    @include media('<wide') {
        padding: 0 0 spacing();
    }
}

.c-footer-heading--button {
    align-items: center;
    background: none;
    border-style: none;
    color: $color-headings;
    display: flex;
    font-weight: $font-weight-headings;
    justify-content: space-between;
    margin: 0;
    padding: spacing(x2);
    text-align: left;
    width: 100%;
    @include font-size($footer-heading-font-size);

    @include media('<wide') {
        cursor: pointer;
    }

    @include media('>=wide') {
        padding: 0;
    }
}

.c-footer-row {
    display: flex;
    flex-flow: column nowrap;
    justify-content: space-between;
    padding: spacing(x2);

    @include media('>=wide') {
        padding: spacing(x4);
        flex-flow: row nowrap;
    }
}

// this modifier allows children in the row to stay on one line
// if they fit together for screens smaller than 1025px
// standard c-footer-row splits children into separate rows for screens smaller than 1025px
.c-footer-row--combined {
    flex-flow: row wrap;

    @include media('>=wide') {
        flex-flow: row nowrap;
    }
}

.c-footer-row--notEqualTopAndBottomPad {
    padding-top: 20px;
    padding-bottom: 40px;
}

.c-footer-row--noPadBelowWide {
    @include media('<wide') {
        padding: 0;
    }
}

.c-footer-row--noBottomPad {
    padding-bottom: 0;
}

.c-footer-row--rightAlignedOnDesktopView {
    justify-content: flex-end;

    @include media('<wide') {
        justify-content: flex-start;
    }
}

.c-footer-list {
    padding: 0;
    list-style: none;
    list-style-image: none;
    margin-top: 0;
    margin-bottom: spacing(x2);
    margin-left: spacing(x2);
    display: flex;
    flex-flow: column nowrap;
    justify-content: flex-start;

    & > li {
        margin-bottom: 0;

        &:before {
            content: none;
        }
    }

    @include media('>=wide') {
        margin: spacing(x2) 0 0 0;
    }

    .is-collapsed & {
        display: none;
    }
}

.c-footer-list--inline {
    margin: 0 0 spacing(x2);
    align-items: center;
    flex-flow: row wrap;

    @include media('>=wide') {
        margin-bottom: 0;
    }
}

.c-footer-list--noBottomMargin {
    margin-bottom: 0;
}

</style>
