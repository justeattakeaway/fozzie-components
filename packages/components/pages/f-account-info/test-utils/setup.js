import Vuex from 'vuex';
import { createLocalVue } from '@vue/test-utils';
import { VueI18n } from '@justeat/f-globalisation';
import tenantConfigs from '../src/tenants';

export const localVue = createLocalVue();
localVue.use(VueI18n);
localVue.use(Vuex);

export const i18n = {
    locale: 'en-GB',
    messages: {
        'en-GB': tenantConfigs['en-GB'].messages
    }
};

export const baseUrl = 'https://smartGatewayBaseUrl.com';

export const token = 'some-auth-token';

export const conversationId = 'b7386108-95e6-4e73-9421-5b066c089153';

export const consumerDetailsGetResponse = {
    data: {
        ConsumerId: 1050450174,
        Id: '8It2lEaJjG82nJtxO00Iab0CoF0=',
        CreatedDate: '2019-04-16T17:25:00.000Z',
        EmailAddress: 'satchin.moriyoka@example.com',
        FirstName: 'Satchin',
        LastName: 'Moriyoka',
        PhoneNumber: '+441234567890',
        DateOfBirth: '1914-04-16T17:25:00.000Z',
        TermsAndConditionsId: 0,
        WantsNewsletter: true,
        AccountType: 'Default',
        ConsumerStatus: 'ReturnConsumer'
    }
};

const consumerAddressGetResponse = {
    AddressId: 1050450174,
    City: 'Bristol',
    ZipCode: 'BS1 4DJ',
    AddressName: 'Office',
    IsDefault: true, // Key field
    Line1: 'Just Eat',
    Line2: 'Broad Quay House'
};

export const consumerAddressesGetResponse = {
    data: [{
        ...consumerAddressGetResponse
    },
    {
        AddressId: 1050450175,
        City: 'Bristol',
        ZipCode: 'BS1 9DJ',
        AddressName: 'Office',
        IsDefault: false,
        Line1: 'Takeaway',
        Line2: 'Broad Quay House'
    }]
};

export const consumerDetailsMappedModel = {
    id: consumerDetailsGetResponse.data.ConsumerId,
    firstName: consumerDetailsGetResponse.data.FirstName,
    lastName: consumerDetailsGetResponse.data.LastName,
    phoneNumber: consumerDetailsGetResponse.data.PhoneNumber,
    emailAddress: consumerDetailsGetResponse.data.EmailAddress
};

export const consumerAddressMappedModel = {
    line1: consumerAddressGetResponse.Line1,
    line2: consumerAddressGetResponse.Line2,
    line3: consumerAddressGetResponse.Line3,
    locality: consumerAddressGetResponse.City,
    postcode: consumerAddressGetResponse.ZipCode
};

export const consumerViewModel = {
    ...consumerDetailsMappedModel,
    ...consumerAddressMappedModel
};

export const consumerStateModel = {
    consumer: {
        ...consumerDetailsMappedModel,
        ...consumerAddressMappedModel
    }
};
