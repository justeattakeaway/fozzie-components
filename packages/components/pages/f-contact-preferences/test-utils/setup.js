export const contactPreferencesGetResponse = {
    data: {
        Preferences: [{
            DisplayName: 'Order status',
            Sort: 1,
            Key: 'orderstatus',
            Push: false,
            Email: true,
            Sms: true
        }, {
            DisplayName: 'Review meal',
            Sort: 2,
            Key: 'reviewmeal',
            Push: false,
            Email: false,
            Sms: false
        }, {
            DisplayName: 'News & offers',
            Sort: 3,
            Key: 'news',
            Push: false,
            Email: false,
            Sms: false
        }
        ],
        PreferenceVersionViewed: 0
    }
};

export const contactPreferencesViewModel = {
    preferenceVersionViewed: 0,
    preferences: [{
        displayName: 'Order status',
        isEmailEnabled: false,
        emailValue: true,
        key: 'orderstatus',
        pushEnabled: false,
        pushValue: false,
        isSmsEnabled: false,
        smsValue: true,
        sort: 1,
        visible: true
    }, {
        displayName: 'Review meal',
        isEmailEnabled: true,
        emailValue: false,
        key: 'reviewmeal',
        pushEnabled: true,
        pushValue: false,
        isSmsEnabled: true,
        smsValue: false,
        sort: 2,
        visible: false
    }, {
        displayName: 'News & offers',
        isEmailEnabled: true,
        emailValue: false,
        key: 'news',
        pushEnabled: true,
        pushValue: false,
        isSmsEnabled: true,
        smsValue: false,
        sort: 3,
        visible: true
    }]
};

export const filteredContactPreferencesModel = [{
    displayName: 'Order status',
    isEmailEnabled: false,
    emailValue: true,
    key: 'orderstatus',
    pushEnabled: false,
    pushValue: false,
    isSmsEnabled: false,
    smsValue: true,
    sort: 1,
    visible: true
}, {
    displayName: 'News & offers',
    isEmailEnabled: true,
    emailValue: false,
    key: 'news',
    pushEnabled: true,
    pushValue: false,
    isSmsEnabled: true,
    smsValue: false,
    sort: 3,
    visible: true
}];

export const contactPreferencesUpdateModel = {
    Preferences: [{
        DisplayName: 'Order status',
        Sort: 1,
        Key: 'orderstatus',
        Push: false,
        Email: true,
        Sms: true
    }, {
        DisplayName: 'Review meal',
        Sort: 2,
        Key: 'reviewmeal',
        Push: false,
        Email: false,
        Sms: false
    }, {
        DisplayName: 'News & offers',
        Sort: 3,
        Key: 'news',
        Push: false,
        Email: false,
        Sms: false
    }],
    DeviceToken: null,
    DeviceType: null,
    PhoneNumber: null,
    PreferenceVersionViewed: 0
};

export const baseUrl = 'https://smartGatewayBaseUrl.com';

export const token = 'some-auth-token';

export const conversationId = 'b7386108-95e6-4e73-9421-5b066c089153';

export const contactPreferencesUpdateBody = {
    Preference: [{
        DisplayName: 'Order status',
        Sort: 1,
        Key: 'orderstatus',
        Push: false,
        Email: true,
        Sms: true
    }, {
        DisplayName: 'Review meal',
        Sort: 2,
        Key: 'reviewmeal',
        Push: false,
        Email: false,
        Sms: false
    }, {
        DisplayName: 'News & offers',
        Sort: 3,
        Key: 'news',
        Push: false,
        Email: true,
        Sms: false
    }
    ],
    DeviceToken: null,
    DeviceType: null,
    PhoneNumber: null,
    PreferenceVersionViewed: 0
};
