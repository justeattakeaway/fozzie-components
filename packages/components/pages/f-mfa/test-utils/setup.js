export const baseUrl = 'https://smartGatewayBaseUrl.com';

export const conversationId = 'b7386108-95e6-4e73-9421-5b066c089153';

// eslint-disable-next-line camelcase
export const postBody = { mfa_token: 'some_mfa_token', otp: 'ABC123' };
