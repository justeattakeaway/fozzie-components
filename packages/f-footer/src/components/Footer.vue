<template>
    <footer
        :data-theme="theme"
        class="c-footer">
        <div
            class="c-footer-container c-footer-row c-footer-row--noPadBelowWide">
            <link-list
                v-for="(linkList, i) in copy.linkLists"
                :key="i + '_ButtonList'"
                :link-list="linkList" />
        </div>

        <div class="c-footer-light">
            <div class="c-footer-container">
                <div
                    v-if="copy.linkButtonList.length && showCourierLinks"
                    data-test-id="courierLinks-wrapper"
                    class="c-footer-row c-footer-row--noBottomPad">
                    <button-list
                        v-for="(buttonList, i) in copy.linkButtonList"
                        :key="i + '_ButtonList'"
                        :button-list="buttonList" />
                </div>

                <div class="c-footer-row">
                    <icon-list
                        :title="copy.downloadOurApps"
                        :icons="copy.appStoreIcons"
                        :locale="copy.locale"
                        is-apps />

                    <feedback-block
                        :title="copy.feedback"
                        :text="copy.improveOurWebsite"
                        :button-text="copy.sendFeedback" />

                    <icon-list
                        :icons="copy.socialIcons"
                        :title="copy.followUs"
                        is-social />
                </div>
            </div>
        </div>

        <div
            class="c-footer-container c-footer-row c-footer-row--combined c-footer-row--notEqualTopAndBottomPad c-footer-row--noPadBelowWide">
            <country-selector
                :current-country-name="copy.currentCountryName"
                :current-country-key="copy.currentCountryKey"
                :countries="copy.countries"
                :change-country-text="copy.changeCurrentCountry" />

            <legal-field
                v-if="metaLegalFieldEnabled"
                :info="copy.metaLegalField" />

            <icon-list
                :icons="copy.paymentIcons"
                is-payments />
        </div>
    </footer>
</template>

<script>
import sharedServices from '@justeat/f-services';
import ButtonList from './ButtonList.vue';
import CountrySelector from './CountrySelector.vue';
import FeedbackBlock from './FeedbackBlock.vue';
import IconList from './IconList.vue';
import LegalField from './LegalField.vue';
import LinkList from './LinkList.vue';
import tenantConfigs from '../tenants';

export default {
    components: {
        ButtonList,
        CountrySelector,
        FeedbackBlock,
        IconList,
        LegalField,
        LinkList
    },

    props: {
        locale: {
            type: String,
            default: ''
        },

        showCourierLinks: {
            type: Boolean,
            default: true
        }
    },

    data () {
        const locale = sharedServices.getLocale(tenantConfigs, this.locale, this.$i18n);
        const localeConfig = tenantConfigs[locale];
        const theme = sharedServices.getTheme(locale);

        return {
            copy: { ...localeConfig },
            theme
        };
    },

    computed: {
        metaLegalFieldEnabled () {
            return Object.keys(this.copy.metaLegalField).length > 0;
        }
    }
};
</script>

<style lang="scss">
.c-footer {
    background-color: $footer-bgColor;
    color: $footer-textColor;
}

.c-footer-light {
    background-color: $footer-bgLight;
}

.c-footer-container {
    width: 100%;
    max-width: #{$layout-max-width}px;
    margin: 0 auto;
}

.c-footer-heading {
    @include font-size(mid);
    padding: spacing(x2);
    padding-left: 0;
    font-family: $font-family-headings;
    font-weight: $font-weight-headings;

    @include theme(ml) {
        font-family: $font-family-headings--ml;
        font-weight: $font-weight-headings--ml;
    }
}

.c-footer-heading--shortBelowWide {
    @include media('<wide') {
        padding: 0 0 spacing();
    }
}

.c-footer-heading--button {
    align-items: center;
    background: none;
    border-style: none;
    color: $color-headings;
    display: flex;
    font-family: $font-family-headings;
    font-weight: $font-weight-headings;
    justify-content: space-between;
    margin: 0;
    padding: spacing(x2);
    text-align: left;
    width: 100%;
    @include font-size(mid);

    @include theme(ml) {
        font-family: $font-family-headings--ml;
        font-weight: $font-weight-headings--ml;
    }

    @include media('<wide') {
        cursor: pointer;
    }

    @include media('>=wide') {
        padding: 0;
    }
}

.c-footer-row {
    display: flex;
    flex-flow: column nowrap;
    justify-content: space-between;
    padding: spacing(x2);

    @include media('>=wide') {
        padding: spacing(x4);
        flex-flow: row nowrap;
    }
}

// this modifier allows children in the row to stay on one line
// if they fit together for screens smaller than 1025px
// standard c-footer-row splits children into separate rows for screens smaller than 1025px
.c-footer-row--combined {
    flex-flow: row wrap;

    @include media('>=wide') {
        flex-flow: row nowrap;
    }
}

.c-footer-row--notEqualTopAndBottomPad {
    padding-top: 20px;
    padding-bottom: 40px;
}

.c-footer-row--noPadBelowWide {
    @include media('<wide') {
        padding: 0;
    }
}

.c-footer-row--noBottomPad {
    padding-bottom: 0;
}

.c-footer-list {
    padding: 0;
    list-style: none;
    list-style-image: none;
    margin-top: 0;
    margin-bottom: spacing(x2);
    margin-left: spacing(x2);
    display: flex;
    flex-flow: column nowrap;
    justify-content: flex-start;

    & > li {
        margin-bottom: 0;

        &:before {
            content: none;
        }
    }

    @include media('>=wide') {
        margin: spacing(x2) 0 0 0;
    }

    .is-collapsed & {
        display: none;
    }
}

.c-footer-list--inline {
    margin: 0 0 spacing(x2);
    align-items: center;
    flex-flow: row wrap;

    @include media('>=wide') {
        margin-bottom: 0;
    }
}

</style>
