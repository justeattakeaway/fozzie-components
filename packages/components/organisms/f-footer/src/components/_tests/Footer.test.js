import { shallowMount } from '@vue/test-utils';
import Footer from '../Footer.vue';

describe('Footer', () => {
    allure.feature('Footer');
    it('should be defined', () => {
        const wrapper = shallowMount(Footer);
        expect(wrapper.exists()).toBe(true);
    });

    it('should render default component markup', () => {
        // Arrange & Act
        const wrapper = shallowMount(Footer);

        // Assert
        expect(wrapper).toMatchSnapshot();
    });

    it('should render ml themed component if AU local passed', () => {
        // Arrange & Act
        const propsData = {
            locale: 'en-AU'
        };
        const wrapper = shallowMount(Footer, { propsData });

        // Assert
        expect(wrapper.attributes('data-theme')).toBe('ml');
    });

    it('should render ml themed component if NZ local passed', () => {
        // Arrange & Act
        const propsData = {
            locale: 'en-NZ'
        };
        const wrapper = shallowMount(Footer, { propsData });

        // Assert
        expect(wrapper.attributes('data-theme')).toBe('ml');
    });

    it('should render je themed component if NO local passed', () => {
        // Arrange & Act
        const propsData = {
            locale: 'nb-NO'
        };
        const wrapper = shallowMount(Footer, { propsData });

        // Assert
        expect(wrapper.attributes('data-theme')).toBe('je');
    });

    it('should not render country selector if `showCountrySelector` is false', () => {
        // Arrange & Act
        const propsData = {
            locale: 'en-GB',
            showCountrySelector: false
        };
        const wrapper = shallowMount(Footer, { propsData });

        // Assert
        expect(wrapper.find('[data-test-id="country-selector"]').exists()).toBe(false);
    });

    it('should render country selector if `showCountrySelector` is true', () => {
        // Arrange & Act
        const propsData = {
            locale: 'en-GB',
            showCountrySelector: true
        };
        const wrapper = shallowMount(Footer, { propsData });

        // Assert
        expect(wrapper.find('[data-test-id="country-selector"]').exists()).toBe(true);
    });
});
