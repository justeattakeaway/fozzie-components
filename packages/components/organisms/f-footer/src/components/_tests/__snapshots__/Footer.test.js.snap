// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer should render default component markup 1`] = `
<footer data-theme="je" data-test-id="footer-component" class="c-footer">
  <div class="c-footer-container c-footer-row c-footer-row--noPadBelowWide">
    <link-list-stub linklist="[object Object]"></link-list-stub>
    <link-list-stub linklist="[object Object]"></link-list-stub>
    <link-list-stub linklist="[object Object]"></link-list-stub>
    <link-list-stub linklist="[object Object]"></link-list-stub>
    <link-list-stub linklist="[object Object]"></link-list-stub>
  </div>
  <div class="c-footer-light">
    <div class="c-footer-container">
      <!---->
      <div class="c-footer-row">
        <icon-list-stub icons="[object Object],[object Object],[object Object]" title="Download our apps" isapps="true" locale="en-GB"></icon-list-stub>
        <feedback-block-stub title="Feedback" text="Help us improve our website" buttontext="Send feedback"></feedback-block-stub>
        <icon-list-stub icons="[object Object],[object Object],[object Object],[object Object],[object Object]" title="Follow us" issocial="true" locale="en-GB"></icon-list-stub>
      </div>
    </div>
  </div>
  <div class="c-footer-container c-footer-row c-footer-row--combined c-footer-row--notEqualTopAndBottomPad c-footer-row--noPadBelowWide">
    <country-selector-stub currentcountryname="United Kingdom" currentcountrykey="gb" countries="[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object]" changecountrytext="You are on the UK website, click here to change." data-test-id="country-selector"></country-selector-stub>
    <!---->
    <icon-list-stub icons="[object Object],[object Object],[object Object]" title="" ispayments="true" locale="en-GB"></icon-list-stub>
  </div>
</footer>
`;
